src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -478,14 +490,11 @@ void SpatialPooler::compute(const UInt inputArray[], bool learn, UInt activeArra
     boostedOverlaps_.assign(overlaps_.begin(), overlaps_.end());
   }
 
-  inhibitColumns_(boostedOverlaps_, activeColumns_);
-  const vector<UInt> spars = VectorHelpers::sparseToBinary<UInt>(activeColumns_, numColumns_);
-  copy(begin(spars), end(spars), activeArray);
-
-
+  inhibitColumns_(boostedOverlaps_, active.getFlatSparse());
+  active.setFlatSparse( active.getFlatSparse() );
```

Comment:
what is the meaning of SDR.setFlatSparse(SDR.getFlatSparse())? Seems identity to me...

Reply:
This tells the SDR that its internal data has changed and that it needs to update its saved format conversations. The line above it modifies the sdr data inplace, and this line tells the sdr that we are done modifying.  I will add an inline comment to the code explaining this line, and maybe rewrite both lines for clarity.